### ISSUE TITLE
unable to build devicetree overlays

### ISSUE THREAD

**Issue by flokli** [NixOS/nixpkgs#169070]
I'm not sure if it never worked with the in-nixpkgs tooling, or if it recently regressed - but I'm unable to build a closure with the following device tree overlay:

```nix
  hardware.deviceTree.overlays = [{
    name = "onboard-wifi";
    dtsText = ''
      /dts-v1/;
      /plugin/;

      / {
        compatible = "pine64,pine64-plus", "allwinner,sun50i-a64";

        fragment@0 {
          target = <&mmc1>;
          __overlay__ {
            pinctrl-names = "default";
            pinctrl-0 = <&mmc1_pins>;
            vmmc-supply = <&reg_dldo4>;
            vqmmc-supply = <&reg_eldo1>;
            mmc-pwrseq = <&wifi_pwrseq>;
            non-removable;
            bus-width = <4>;
            status = "okay";
          };
        };

        wifi_pwrseq: wifi_pwrseq {
          compatible = "mmc-pwrseq-simple";
          reset-gpios = <&r_pio 0 2 1>; /* PL2 */
                                  //^ GPIO_ACTIVE_LOW
        };
      };
    '';
  }];
```

The derivation fails to build with the following error message:

```
Error at 'compatible': FDT_ERR_NOTFOUND
error: build of '/nix/store/9kxhd9z88q5fi991x49fc2lskrbcysid-device-tree-overlays.drv' on 'ssh://flokli@aarch64.nixos.community' failed: error: builder for '/nix/store/9kxhd9z88q5fi991x49fc2lskrbcysid-device-tree-overlays.drv' failed with exit code 1;
              last 10 log lines:
              > './hisilicon/hi6220-hikey.dtb' -> '/nix/store/zspkrzkfvbp66m48p1r7nnb0sdqi78jz-device-tree-overlays/./hisilicon/hi6220-hikey.dtb'
              > './hisilicon/hip05-d02.dtb' -> '/nix/store/zspkrzkfvbp66m48p1r7nnb0sdqi78jz-device-tree-overlays/./hisilicon/hip05-d02.dtb'
              > ./socionext -> /nix/store/zspkrzkfvbp66m48p1r7nnb0sdqi78jz-device-tree-overlays/./socionext
              > './socionext/uniphier-ld20-global.dtb' -> '/nix/store/zspkrzkfvbp66m48p1r7nnb0sdqi78jz-device-tree-overlays/./socionext/uniphier-ld20-global.dtb'
              > './socionext/uniphier-ld11-global.dtb' -> '/nix/store/zspkrzkfvbp66m48p1r7nnb0sdqi78jz-device-tree-overlays/./socionext/uniphier-ld11-global.dtb'
              > './socionext/uniphier-ld20-ref.dtb' -> '/nix/store/zspkrzkfvbp66m48p1r7nnb0sdqi78jz-device-tree-overlays/./socionext/uniphier-ld20-ref.dtb'
              > './socionext/uniphier-pxs3-ref.dtb' -> '/nix/store/zspkrzkfvbp66m48p1r7nnb0sdqi78jz-device-tree-overlays/./socionext/uniphier-pxs3-ref.dtb'
              > './socionext/uniphier-ld20-akebi96.dtb' -> '/nix/store/zspkrzkfvbp66m48p1r7nnb0sdqi78jz-device-tree-overlays/./socionext/uniphier-ld20-akebi96.dtb'
              > './socionext/uniphier-ld11-ref.dtb' -> '/nix/store/zspkrzkfvbp66m48p1r7nnb0sdqi78jz-device-tree-overlays/./socionext/uniphier-ld11-ref.dtb'
              > Error at 'compatible': FDT_ERR_NOTFOUND
              For full logs, run 'nix log /nix/store/9kxhd9z88q5fi991x49fc2lskrbcysid-device-tree-overlays.drv'.
```

Related: https://github.com/NixOS/nixpkgs/pull/79370

cc @leifhelm @sorki @kwohlfahrt @samueldr

**Comment by flokli** [comment]
It seems the "compatibility" check in https://github.com/NixOS/nixpkgs/pull/79370 isn't sufficient, and the tooling tries to apply overlays to devicetree files that are not compatible.

If I set a filter with `hardware.deviceTree.filter` that only includes (a) compatible dts, it builds.

**Comment by sorki** [comment]
Hi, this is a regression as some device tree files are missing `compatible` stanza.  Fixed in #166114

**Comment by sorki** [comment]
Resolved in https://github.com/NixOS/nixpkgs/pull/205595
